sgx-types/src/ti.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Report Target Info (Section 38.16)
+//! The Target Info is used to identify the target enclave that will be able to cryptographically
+//! verify the REPORT structure returned by the EREPORT leaf. Must be 512-byte aligned.
+
+use super::{attr::Attributes, misc::MiscSelect, secs::Secs};
+
+/// Table 38-22
+#[repr(C, align(512))]
+pub struct TargetInfo {
+    /// MRENCLAVE of the target enclave.
+    pub mrenclave: [u8; 32],
+    /// Attributes of the target enclave.
+    pub attributes: Attributes,
+    reserved0: u32,
+    /// MiscSelect of the target enclave.
+    pub misc: MiscSelect,
+    reserved1: [u8; 57],
+}
+
+impl TargetInfo {
+    /// Creates new TargetInfo from an enclave's SECS.
+    pub const fn new(secs: Secs) -> Self {
+        Self {
+            mrenclave: secs.mrenclave,
+            attributes: secs.attr,
+            reserved0: 0u32,
+            misc: secs.misc,
+            reserved1: [0u8; 57],
+        }
+    }
+}
+
+impl AsRef<[u8]> for TargetInfo {
+    fn as_ref(&self) -> &[u8] {
+        unsafe {
+            core::slice::from_raw_parts(
+                self as *const Self as *const u8,
+                core::mem::size_of_val(self),
+            )
+        }
+    }
+}
```

Comment:
Don't `impl AsRef<[u8]>` if it isn't used.

Reply:
I thought we might use it to send `Targetinfo` between enclaves; but I can remove it if it is unnecessary.

Reply:
I think we will want proper serialization (i.e. serde) for that in the future.